pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -204,26 +210,30 @@ func (gc *GarbageCollector) Sync(discoveryClient discovery.ServerResourcesInterf
 
 		// Once we get here, we should not unpause workers until we've successfully synced
 		attempt := 0
-		wait.PollImmediateUntil(100*time.Millisecond, func() (bool, error) {
+		wait.PollImmediateUntilWithContext(ctx, 100*time.Millisecond, func(ctx context.Context) (bool, error) {
 			attempt++
 
 			// On a reattempt, check if available resources have changed
 			if attempt > 1 {
 				newResources = GetDeletableResources(discoveryClient)
 				if len(newResources) == 0 {
-					klog.V(2).Infof("no resources reported by discovery (attempt %d)", attempt)
+					logger.V(2).Info("no resources reported by discovery", "attempt", attempt)
```

Comment:
Strictly speaking, this would need to look up the logger again because it runs with a different `ctx` than the surrounding function. But in practice it probably doesn't matter, so let's leave it.